snippets/language-jolie.cson

```diff
@@ -0,0 +1,140 @@
+'.source.jolie':
+  'outputPort':
+    'prefix': 'outputPort'
+    'body': 'outputPort ${1:portName} {\n\tLocation: $2\n\tProtocol: $3\n\tInterfaces: $4\n}'
+  'inputPort':
+    'prefix': 'inputPort'
+    'body': 'inputPort ${1:PortName} {\n\tLocation: $2\n\tProtocol: $3\n\tInterfaces: $4\n}'
+  'Interface':
+    'prefix': 'interface'
+    'body': 'interface ${1:interfaceName} {\n\tRequestResponse: $2\n\tOneWay: $3\n}'
+  'main':
+    'prefix': 'main'
+    'body': 'main {\n\t$0 \n}'
+  'init':
+    'prefix': 'init'
+    'body': 'init {\n\t$0 \n}'
+  'constants':
+    'prefix': 'const'
+    'body': 'constants {\n\t${1:constant} = ${2:value}\n}'
+  'include':
+    'prefix': 'include'
+    'body': 'include "${1:file}.iol"'
+  'with':
+    'prefix': 'with'
+    'body': 'with( ${1:parent} ){\n\t.${2:child}\n}'
+  'while':
+    'prefix': 'while'
+    'body': 'while( ${1:condition} ) {\n\t$2\n}'
+  'undef':
+    'prefix': 'undef'
+    'body': 'undef( ${0:variableName} )'
+  'type (custom)':
+    'prefix': 'type'
+    'body': 'type ${1:typeName}: ${2:Type} {\n\t.${3:subNodeName}: ${4:Type}\n}'
+  'type (choice)':
+    'prefix': 'type'
+    'body': 'type ${0:typeName}: ${1:Type1} | ${2:Type2} | ${3:Type3}'
+  'type (basic)':
+    'prefix': 'type (basic)'
+    'body': 'type ${1:typeName}: ${2:basicType}'
+  'throws':
+    'prefix': 'throws'
+    'body': 'throws ${1:faultName}( ${2:faultType} )'
+  'throw':
+    'prefix': 'throw'
+    'body': 'throw( ${1:error} )'
+  'synchronized':
+    'prefix': 'synchronized'
+    'body': 'synchronized( ${1:token} ){\n\t$2\n}'
+  'spawn - parallel composition of output processes':
+    'prefix': 'spawn'
+    'body': '// N.B. the "in" clause is optional and eases
+// the retrieval of in-process local values. The local variable at\n// position i can be accessed by referring to the i-th element in\n// localVector variable. E.g., if in the spawn block process i sets\n// the value of local variable localVector.v, the value of v\n// is retrieved outside the spawn with localVector[i].v.\nspawn ( ${1:index} over ${2:integerExpression} ) ${3:in ${4:localVector}} {\n\t$5\n}'
+  'socket (localhost)':
+    'prefix': 'socket'
+    'body': 'socket://${1:localhost}:${2:portNumber}'
```

Comment:
This is copied from the sublime-snippet file, but wouldn't it make more sense to put quotation marks around it?